src/sirocco/parsing/_yaml_data_models.py

```diff
@@ -25,46 +26,67 @@
 
 
 class _NamedBaseModel(BaseModel):
-    """Base class for all classes with a key that specifies their name.
-
-    For example:
+    """
+    Base model for reading names from yaml keys *or* attributes.
```

Comment:
Thank you I think this is more understandable already. I still have a suggestion, I must say this is quite pedantic, but I had some troubles to understand the word "attribute": If I understand the [yaml specification](https://yaml.org/spec/1.2.2/#nodes), they dont use the terminology "attributes" but "values".

```suggestion
    Base model for reading names from yaml keys *or* values.
```

Reply:
I should actually have mentioned "keyword args to the constructor", since that was the real goal.